app/src/main/java/com/example/whatsapp_clone/Views/Fragments/RegisterFragment.java

```diff
@@ -1,38 +1,203 @@
+            if (result.getResultCode() == RESULT_OK) {
+                Intent data = result.getData();
+                if (data != null) {
+                    Uri imageUri = data.getData(); // retrieve the URI of the selected image
+                    handleImageSelection(imageUri);
+                }
+            }
+        });
+
+        uploadPictureBtn.setOnClickListener(v -> {
+            uploadPictureBtn.setText(R.string.change_profile_picture);
+            Intent intent = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+            imagePickerLauncher.launch(intent);
+        });
+
+
+        profileImageView.setOnClickListener(v -> {
+            Intent intent = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+            imagePickerLauncher.launch(intent);
+        });
+
+
+        // making the error msgs disappear when clicking on the edit text:
+        binding.emailInput.setOnClickListener(v -> {
+            binding.emailInputLayout.setError(null);
+        });
+        binding.emailInputLayout.setOnClickListener(v -> {
+            binding.emailInputLayout.setError(null);
+        });
+
+        binding.displayNameInput.setOnClickListener(v -> {
+            binding.displayNameInputLayout.setError(null);
+        });
+        binding.displayNameInputLayout.setOnClickListener(v -> {
+            binding.displayNameInputLayout.setError(null);
+        });
+
+        binding.passwordInput.setOnClickListener(v -> {
+            binding.passwordInputLayout.setError(null);
+        });
+        binding.passwordInputLayout.setOnClickListener(v -> {
+            binding.passwordInputLayout.setOnClickListener(null);
+        });
+
+        binding.passwordConfirmationInput.setOnClickListener(v -> {
+            binding.passwordConfirmationInputLayout.setError(null);
+        });
+        binding.passwordConfirmationInputLayout.setOnClickListener(v -> {
+            binding.passwordConfirmationInputLayout.setError(null);
+        });
+
+        // Return the root view
+        return rootView;
+    }
+
+
+    private void handleImageSelection(Uri imageUri) {
+        try {
+            Bitmap bitmap = MediaStore.Images.Media.getBitmap(requireActivity().getContentResolver(), imageUri);
+            profileImageView.setImageBitmap(bitmap);
+            base64ProfilePic = convertBitmapToBase64(bitmap);
+            profileImageView.setVisibility(View.VISIBLE);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+    private String convertBitmapToBase64(Bitmap bitmap) {
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 100, byteArrayOutputStream);
+        byte[] byteArray = byteArrayOutputStream.toByteArray();
+        return Base64.encodeToString(byteArray, Base64.DEFAULT);
     }
 
+
```

Comment:
the viewmodel should get the input from the user when click the button and check all the needed info there. this is not code that should be in the fragment itself